besu/src/main/java/org/hyperledger/besu/cli/options/stable/DataStorageOptions.java

```diff
@@ -95,8 +95,7 @@ public class DataStorageOptions implements CLIOptions<DataStorageConfiguration>
 
   @Option(
       names = "--receipt-compaction-enabled",
-      description = "Enables compact storing of receipts (default: ${DEFAULT-VALUE}).",
-      arity = "1")
+      description = "Enables compact storing of receipts (default: ${DEFAULT-VALUE})")
```

Comment:
you could also set `arity = "0..1"` so it works with or without the boolean parameter

Reply:
In the docs of the version we're using: https://javadoc.io/doc/info.picocli/picocli/4.7.5/picocli/CommandLine.Option.html#arity--

>  By default, flags (boolean options) have arity "0..1",

Not sure if this once used to work, but the behaviour I see is that a valueless option now acts as a toggle. So if we enable it by default, specifying `--receipt-compaction-enabled` toggles and therefore disables the feature 😓  

https://picocli.info/#_toggle_boolean_flags
> When a flag option is specified on the command line picocli will set its value to the opposite of its default value.

Specifying a `fallBackValue` was the only way I could get this to work as intended.

Reply:
I've changed to use `fallBackValue="true"`. I forgot to push the updated changes on Friday. I think it's working now.